### PR TITLE
Improve dependencies

### PR DESCRIPTION
**Pull request by mpapierski**
This is related to #6 where some dependencies were out of date and that could block `althea_rs` from integration guac.

This also reduces cross dependencies to minimum i.e. guac depends on althea_types for eth-types aliases, and also guac depends on eth-types directly. Althea depends on guac, and althea_types and that created some problems in integration.

Additionally I believe a significant change was made to remove `Cargo.lock` as guac is a library and according to https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries this is the right way to do it.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -2,5 +2,6 @@
 members = ["guac_core", "guac_actix", "guac_http"]
 
 [patch.crates-io]
-ethereum-types = {git="https://github.com/paritytech/primitives.git"}
-actix = { git = "https://github.com/kingoflolz/actix.git", branch = "althea-mesh" }
\ No newline at end of file
+ethereum-types = {git="https://github.com/paritytech/primitives.git", rev="6f5f681"}
```
Comment (jkilpatr) [inline Cargo.toml]: why are we still using the git version at all?

File: Cargo.toml [hunk 2]
```diff
@@ -2,5 +2,6 @@
...
+actix = { git = "https://github.com/kingoflolz/actix.git", branch = "althea-mesh" }
```
Comment (jkilpatr) [inline Cargo.toml]: likewise this can be cargo now

File: guac_actix/Cargo.toml [hunk 3]
```diff
@@ -9,12 +9,12 @@ actix-web = {version = "0.7",  default-features = false}
...
+althea_types = "0.1.0"
 log = "0.4"
 serde = "1.0.0"
 serde_derive = "1.0.0"
-serde_json = "1.0.0"
+serde_json = "1.0.24"
 qutex = "^0.2"
 tokio = "0.1"
 bytes = "0.4"
+ethereum-types = "0.3.2"
```
Comment (jkilpatr) [inline guac_actix/Cargo.toml]: this is just what's available on cargo... should be trivial to remove the git patch

File: Cargo.toml (hunk 1 above)
Comment (mpapierski) [reply Cargo.toml]: I gave it a try again and it works with pin, at least from guac side.

File: Cargo.toml (hunk 2 above)
Comment (mpapierski) [reply Cargo.toml]: Agreed just forgot about this one

File: guac_actix/Cargo.toml (hunk 3 above)
Comment (mpapierski) [reply guac_actix/Cargo.toml]: :+1: fixed. Additionally I found out `owning_ref` could be used from upstream.

**Pull request merged**